src/components/progressbar.tsx

```diff
@@ -17,21 +17,46 @@
  */
 
 import "../css/progressbar.css";
-import React from "react";
+import React, { useContext } from "react";
+import { ConfigContext } from "../config";
+import { Status } from "../rpc/transmission";
 
 interface ProgressBarProps {
     now: number,
     max?: number,
     label?: string,
     animate?: boolean,
+    status?: number,
```

Comment:
Status logic should not be in ProgressBar component because it is generic. It should have an enum variant prop that gives it correct color, determine the variant in parent component that knows what status means, let progressbar handle the color based on variant.

Reply:
Kind of confused on this one. Do you mean the ProgressBar component should have an enum for the possible colors, and everything that calls ProgressBar should manually convert e.g. the torrent status into a ProgressBarColor? Which would mean repeating the code between torrenttable.tsx and details.tsx?

Reply:
No need to repeat the code, extract it into a helper function, you can put it in `torrent.ts`.